src/scripts/main.js

```diff
@@ -1,3 +1,38 @@
 'use strict';
 
-// write code here
+const headers = document.querySelectorAll('th');
+const tBody = document.querySelector('tBody');
+const rows = document.querySelectorAll('tBody tr');
+
+const array = [...rows];
+
+headers.forEach((header) => {
+  header.onclick = function (e) {
+    const index = e.target.cellIndex;
+
+    sortBy(array, index);
+  };
+});
+
+function sortBy(arrayOf, index) {
+  arrayOf.sort((td1, td2) => {
+    if (index === 0 || index === 1) {
+      return td1.children[index].innerHTML.localeCompare(
+        td2.children[index].innerHTML,
+      );
+    } else {
+      return (
+        getNumber(td1.children[index].innerHTML) -
+        getNumber(td2.children[index].innerHTML)
+      );
+    }
```

Comment:
The sorting logic is hard-coded to handle only three cases: when the index is 0 or 1, and when it's not. This makes the function less flexible and harder to maintain. Consider refactoring this part to make it more generic. For example, you could pass a comparator function as an argument to the `sortBy` function.